packages/localize/src/date/parseDate.js

```diff
@@ -53,10 +53,11 @@ export function parseDate(dateString) {
     default:
       parsedString = '0000/00/00';
   }
+
   const parsedDate = new Date(parsedString);
-  // Check if parsedDate is not `Invalid Date`
+  // Check if parsedDate is not `Invalid Date` or that the date has changed (e.g. the not existing 31.02.2020)
   // eslint-disable-next-line no-restricted-globals
-  if (!isNaN(parsedDate.getTime())) {
+  if (!isNaN(parsedDate.getTime()) && parsedDate.getDate() === Number(parsedString.slice(8, 10))) {
```

Comment:
`Number(parsedString.slice(8, 10)` seems to be based on DateFormatBasedOnLocale `year-month-day`. On a locale with `month-day-year`, it would fail because the day would be somewhere else inside parsedString?

Reply:
In the function above the code (`parsedString `) is always formatted in the same order `year-month-day`.

Reply:
That would either mean [getDateFormatBasedOnLocale](https://github.com/ing-bank/lion/blob/52bf194d3961db24b4a896d0c1d2f500fb1e0385/packages/localize/src/date/getDateFormatBasedOnLocale.js#L8) always gives back `year-month-day` or the switch statement evaluates values it could never get :)

I think we should run `return undefined when no valid date provided` for a different locale as well:

```js
    expect(parseDate('31.02.2020')).to.equal(undefined);
    expect(parseDate('02.31.2020')).to.equal(undefined); // probably fails
```

Reply:
I added those tests (and some extra) all succeeded

Reply:
That's true, I debugged it.
The reason is that `new Date('2020/31/02')` gives back null and `new Date('2020/02/31')` does not apparently :)

Ok, I approve this, but an explanation that `parsedDate.getDate() === Number(parsedString.slice(8, 10))` can only get hit with `year-month-day` due to the platform handling different constructor formats differently, wouldn't hurt imho.

And those extra tests to confirm this (with locales with different orders) would then help as well.